ts-framework/create-function/gram-template-gram/tsconfig.json

```diff
@@ -0,0 +1,36 @@
+{
+  "include": ["src"],
+  "exclude": ["node_modules"],
+  "compilerOptions": {
+    "noEmit": true,
+    "rewriteRelativeImportExtensions": true,
+
+    "module": "nodenext",
+    "target": "esnext",
+    "lib": ["esnext"],
+    "types": ["node"],
+    "resolveJsonModule": true,
+
+    "sourceMap": true,
+    "declaration": true,
+    "declarationMap": true,
+
+    "noUncheckedIndexedAccess": true,
+
+    "noImplicitReturns": true,
+    "noImplicitOverride": true,
+    "noUnusedLocals": true,
+    "noUnusedParameters": true,
+    "noFallthroughCasesInSwitch": true,
+    "noPropertyAccessFromIndexSignature": true,
+
+    "strict": true,
+    "erasableSyntaxOnly": true,
```

Comment:
Are the opinions in this tsconfig all from your own forays into the typescript compiler options? Or are they inspired by some other source?

This is an amazing tsconfig and I would like to know from where the opinions in it arose. Definitely stealing this for future projects.